src/compiler/checker.ts

```diff
@@ -8704,8 +8717,18 @@ namespace ts {
             return widenTypeForVariableLikeDeclaration(getTypeForVariableLikeDeclaration(declaration, /*includeOptionality*/ true), declaration, reportErrors);
         }
 
+        function isGlobalSymbolConstructor(node: Node) {
+            const symbol = getSymbolOfNode(node);
+            const globalSymbol = getGlobalESSymbolConstructorTypeSymbol(/*reportErrors*/ false);
+            return globalSymbol && symbol && symbol === globalSymbol;
+        }
+
         function widenTypeForVariableLikeDeclaration(type: Type | undefined, declaration: any, reportErrors?: boolean) {
             if (type) {
+                // TODO: Remove the following SymbolConstructor special case when back compat with pre-3.0 libs isn't required

```

Comment:
Realistically maybe this isn't a "TODO" and more of a wishful thought - the compat behavior is probably a permanent thing, given our ecosystem.